### PR TITLE
CAS2-326: Re-point Status Updates to Assessments

### PR DESCRIPTION
**Pull request by Harriethw**
# Context
**This cannot be merged until this API work is done https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1840**
[JIRA ticket](https://dsdmoj.atlassian.net/browse/CAS2-326)

# Changes in this PR

This is part of a larger refactoring project to move items concerned with assessing an Application to an Assessment model. https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/main/doc/architecture/decisions/0021-introduce-cas2-assessments.md 

This PR does three main things
1. starts to use the Assessment ID to create a new Status Update - on the service, client and controller levels we move the functions to the Assessment side of things.
2. gets the Status Updates via an Assessment rather than Application - this happens on a couple of controller methods, ensures that any time we need a Status Update it's done via the Assessment.
3. Removes unused functions

(Plus some tidying up of unit tests and config)

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Comment by danhumphreys-moj** [comment]
I'm sure you have done it but do all e2e, unit and integration tests run for this branch when pointing at the `cas2-326-assessments` branch of the API?

File: script/generate-types [hunk 1]
```diff
@@ -8,7 +8,7 @@ cd "$(dirname "$0")/.."
 
 echo "==> Generating types..."
 
-npx openapi-typescript-codegen -i https://raw.githubusercontent.com/ministryofjustice/hmpps-approved-premises-api/main/src/main/resources/static/codegen/built-cas2-api-spec.yml -o ./server/@types/shared -c axios --exportServices false --exportCore false --useUnionTypes true
+npx openapi-typescript-codegen -i https://raw.githubusercontent.com/ministryofjustice/hmpps-approved-premises-api/cas2-326-status-updates-assessments/src/main/resources/static/codegen/built-cas2-api-spec.yml -o ./server/@types/shared -c axios --exportServices false --exportCore false --useUnionTypes true
```
Comment (danhumphreys-moj) [inline script/generate-types:11]: Do we want to leave this in?

File: server/controllers/staticController.test.ts [hunk 2]
```diff
@@ -31,4 +31,13 @@ describe('StaticController', () => {
       expect(response.render).toHaveBeenCalledWith('static/privacy-notice')
     })
   })
+
```
Comment (danhumphreys-moj) [inline server/controllers/staticController.test.ts:34]: Is this related to another ticket?

File: server/controllers/staticController.test.ts [hunk 3]
```diff
@@ -31,4 +31,13 @@ describe('StaticController', () => {
...
+  describe('cookies policy page', () => {
```
Comment (danhumphreys-moj) [inline server/controllers/staticController.test.ts:35]: Is this related to another ticket?

Comment (Harriethw) [reply server/controllers/staticController.test.ts:35]: yes - I've put in the commit message and PR description, just doing some tidying up to add unit test coverage before starting

**Comment by danhumphreys-moj** [comment]
I see you haven't requested a review from me but have looked in any case @Harriethw.  It's a large PR and feel I don't really have anything constructive to comment, other than it looks ok to me!

File: script/generate-types (hunk 1 above)
Comment (Harriethw) [reply script/generate-types:11]: nope, will drop it once the PR is ready to merge, and the API has been done

**Review by patrickjfl — APPROVED** [review]
Looks good to me! And nice job with the extra test coverage etc 🙌
